src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
+     * @param navigationLibrary {NavigationLibrary}
+     * @chainable
      */
-    init({ altitude, category, model, route }, initialRunwayAssignment) {
-        this._setCurrentPhaseFromCategory(category);
-        this._setInitialRunwayAssignmentFromCategory(category, initialRunwayAssignment);
-
-        this.flightPlanAltitude = altitude;
+    init({ altitude, category, destination, model, nextFix, origin, routeString }, navigationLibrary) {
+        this._navigationLibrary = navigationLibrary;
+        this._routeModel = new RouteModel(this._navigationLibrary, routeString);
 
-        if (category === FLIGHT_CATEGORY.DEPARTURE) {
-            this.flightPlanAltitude = model.ceiling;
-        }
+        this._verifyRouteContainsMultipleWaypoints();
+        this._initializeFlightPhaseForCategory(category);
+        this._initializeDepartureAirport(origin);
+        this._initializeDepartureRunway();
+        this._initializeArrivalAirport(destination);
+        this._initializeArrivalRunway();
+        this._initializeFlightPlanAltitude(altitude, category, model);
+        this._initializePositionInRouteToBeginAtFixName(nextFix, category);
 
-        this.legCollection = this._buildLegCollection(route);
+        return this;
     }
 
     /**
-     * Destroy the instance and reset properties
+     * Reset class properties
      *
      * @for LegModel
-     * @method destroy
+     * @method reset
+     * @chainable
      */
-    destroy() {
-        this._navigationLibrary = null;
-        this._previousRouteSegments = [];
+    reset() {
+        this.arrivalAirportModel = null;
+        this.arrivalRunwayModel = null;
         this.currentPhase = '';
+        this.departureAirportModel = null;
         this.departureRunwayModel = null;
-        this.arrivalRunwayModel = null;
         this.flightPlanAltitude = INVALID_NUMBER;
-        this.legCollection = [];
-    }
+        this._routeModel = null;
+        this._navigationLibrary = null;
 
-    /**
-     * Return an array of waypoints in the flight plan that have altitude restrictions
-     *
-     * @for Fms
-     * @method getAltitudeRestrictedWaypoints
-     * @return {array<WaypointModel>}
-     */
-    getAltitudeRestrictedWaypoints() {
-        return this.waypoints.filter((waypoint) => waypoint.hasAltitudeRestriction);
+        return this;
     }
 
     /**
-     * Collects the `.getProcedureTopAltitude()` value from each `LegModel`
-     * in the `#legCollection`, then finds and returns the highest value
+     * Initialize #arrivalAirportModel
      *
-     * @for LegModel
-     * @method getTopAltitude
-     * @return {number}
+     * @for Fms
+     * @method _initializeArrivalAirport
+     * @param destinationIcao {string} ICAO identifier specified by spawn pattern
+     * @private
      */
-    getTopAltitude() {
-        const maxAltitudeFromLegs = _map(this.legCollection, (leg) => leg.getProcedureTopAltitude());
+    _initializeArrivalAirport(destinationIcao) {
```

Comment:
`destinationIcao` is confusing here.  why not just call it `airportIco`?  The method name already provides enough context for what is going on so I feel this parameter can have a more generic name.